### PR TITLE
Upgrade template, add python 3.12 support

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

Merge template updates, notably adding python 3.12 support

Not adding the `.rst` file checker, because the `CHANGLOG.rst` is at top level. It will be added once docs are standardized in this lib.

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-bloom/blob/main/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -42,34 +39,67 @@ common: &common
...
-          python -m pip install tox
-    - run:
-        name: run tox
-        command: python -m tox run -r
-    - save_cache:
-        paths:
-          - .tox
-        key: cache-v1-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "setup.py" }}-{{ checksum "tox.ini" }}
+          sudo apt-get update
+          sudo apt-get install latexmk tex-gyre texlive-fonts-extra
```
Comment (pacrob) [inline .circleci/config.yml:96]: allowing some settings for a full docs build to be added, as will be standardizing the docs in this lib soon.

**Review by kclowes — APPROVED** [review]
Looks good to me! 🚢

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -43,3 +48,8 @@ repos:
     hooks:
     -   id: mypy
         exclude: tests/
+-   repo: https://github.com/PrincetonUniversity/blocklint
+    rev: v0.2.5
+    hooks:
+    -   id: blocklint
+        exclude: 'docs/Makefile|CHANGELOG.rst|tox.ini|41.internal.rst'
```
Comment (reedsa) [inline .pre-commit-config.yaml:55]: Is that `41.internal.rst` needed?

**Review by reedsa — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (pacrob) [reply .pre-commit-config.yaml:55]: That's for `blocklint`. I figured I'll be coming back through adding the normal `release_notes` docs process here in a bit.

**Pull request merged**